### DISCUSSION TITLE
What is a simple way to close viem client?

### DISCUSSION THREAD

**Discussion by crystalin** [wevm/viem#620] [Question]
When not closed, the viem client keeps the socket open which prevents node from finishing.
Currently one solution is:
```
await (viem.transport.getSocket()).close()
```
Is there a better way to achieve that ?

**Comment by anandsuresh** [comment]
Even the aforementioned solution does not work on my end. For some reason, the node process hangs waiting for release of resources.

**Comment by tonisives** [comment]
I want to say that viem caused a memory leak in my app. I wasn't aware that viem opens a socket, and therefore thought I could dispose and recreate instances.

<img width="1347" height="313" alt="Screenshot 2025-11-07 at 05 32 20" src="https://github.com/user-attachments/assets/7516c6bb-66b9-4899-a321-b10e045ed1db" />


There should be a close function, or auto close functionality. I would actually assume that viem uses HTTP requests by default, which don't keep active connection alive indefinitely.



My use case: I used 3 separate RPC-s for queries. For every query, I created a new instance. The previous instance didn't have any pointers, but didn't get GC-d.
After creating instance pooling, the memory leak disappeard.
